brainframe_qt/ui/resources/settings.py

```diff
@@ -1,130 +1,75 @@
         self._cache = None
 
+        if prev_value is not None:
+            self.value_changed.emit(None)
+            self.value_deleted.emit()
+
+
+class SettingsManager(QObject):
+    value_changed = pyqtSignal(str, object)
+    value_deleted = pyqtSignal(str)
+
+    def __init__(self):
+        super().__init__()
 
-class Topic(Enum):
-    CHANGED = "changed"
-
-
-# TODO: Having to pass in a settings_object is less than ideal
-class QSettingsConfig:
-
-    def __init__(self, settings_object):
-        super().__setattr__("_settings", settings_object)
-        super().__setattr__("_overrides", {})
-
-    def __getattribute__(self, item: str) -> Any:
-        if item not in typing.get_type_hints(type(self)):
-            return super().__getattribute__(item)
-
-        _overrides = super().__getattribute__("_overrides")
-        try:
-            return _overrides[item]
-        except KeyError:
-            _settings = super().__getattribute__("_settings")
-            return getattr(_settings, item).val()
-
-    def __setattr__(self, key: str, value: Any):
-        if key not in typing.get_type_hints(type(self)):
-            raise AttributeError(f"Unknown option '{key}'")
-
-        _overrides = super().__getattribute__("_overrides")
-        _overrides[key] = value
-
-    def save_to_disk(self):
-        _settings = super().__getattribute__("_settings")
-        _overrides = super().__getattribute__("_overrides")
-
-        for key, value in _overrides.items():
-            settings_item: Setting = getattr(_settings, key)
-            settings_item.set(value)
-
-
-# System configuration settings
-server_url = Setting(
-    "http://localhost", type_=str, name="server_url"
-)
-server_username = Setting(
-    None, type_=str, name="server_username"
-)
-server_password = Setting(
-    None, type_=bytes, name="server_password"
-)
-# Defines what timezone the user wants times to be displayed in. If this value
-# is an empty string, the current system timezone will be selected.
-user_timezone = Setting(
-    "", type_=str, name="user_timezone"
-)
-
-# License settings
-client_license_accepted = Setting(
-    False, type_=bool, name="client_license_accepted"
-)
-client_license_md5 = Setting(
-    None, type_=str, name="client_license_md5"
-)
-
-# The maximum size of the frame buffer used for all streams
-frame_buffer_size = Setting(
-    300, type_=int, name="frame_buffer_size"
-)
-
-
-def get_user_timezone() -> Timezone:
-    """Gets the timezone the user should be displayed times in as a Pendulum
-    timezone. By default, this is the system timezone.
-
-    :return: The currently configured timezone
-    """
-    if user_timezone.val() == "":
-        # Default to the current timezone
-        return pendulum.now().timezone
-    else:
-        return pendulum.timezone(user_timezone.val())
+        # Search for the descriptor on the class
+        # https://stackoverflow.com/a/3681323/8134178
+        for cls in [self] + self.__class__.mro():
```

Comment:
I'm just curious- was any of this new system inspired by your experience on strawberry?

Reply:
No not at all, unfortunately. I really wanted to be able to use dataclass-style syntax (ala
```python
RenderSettings(SettingsManager):
    draw_lines: bool = False
```
instead of having to use the `Setting` object. It was making me think Strawberry should abstract that part of its code and make it available for libraries